### PR TITLE
chore(#6) - Create simple node script for hangout chat bot

### LINKED ISSUE #6: Create simple node script for hangout chat bot
Expected functions
- Create node script which fetches jira information
- Receive messages and other kinds of events generated by Google Chat
- Send event responses and other messages into Google Chat

### PR DESCRIPTION
**Pull request by hybridx**
Details 
- Changed the bot to support basic commands
   - Example
```
@JIRABot <issue-number> <issue-number>
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,24 +1,127 @@
...
   res.send({ hello: 'there!' })
 })
 
-app.listen(PORT, () => `Listening at http://localhost:${PORT}`)
+app.post('/', async (req, res) => {
+  try {
+    if (req.body && req.body.type) {
+      switch(req.body.type) {
+        case 'ADDED_TO_SPACE':
+          return res.json({ 'text': `Thank you for adding me. Command \help` });
```
Comment (gautamkrishnar) [inline index.js]: You cant use /help to recieve the help message. I guess this should be just `help`

Comment (hybridx) [reply index.js]: Alright makes sense

Comment (hybridx) [reply index.js]: Added the change

File: README.md [hunk 2]
```diff
@@ -1,2 +1,31 @@
-# jbot
-Red Hat Jira Bot
+# JIRA Bot 👋
+
+A jira bot which fetches JIRA issues for you and a bunch of other stuff
```
Comment (diwanshi) [inline README.md]: A bot that retrieves the JIRA issue details from your JIRA server.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,31 @@
...
+  
+1. Set up the .env file with the following values
+- example
+```
+PORT=8080
+JIRA_HOST=developer.atlassian.com
+JIRA_USER=serviceAccount
+JIRA_PASS=serviceAccountPassword
+```
+2. Install libraries using
```
Comment (diwanshi) [inline README.md]: update it to just Installation

File: README.md [hunk 4]
```diff
@@ -1,2 +1,31 @@
...
+PORT=8080
+JIRA_HOST=developer.atlassian.com
+JIRA_USER=serviceAccount
+JIRA_PASS=serviceAccountPassword
+```
+2. Install libraries using 
+```sh
+npm install
+```
+3. Run the for development.</br>
```
Comment (diwanshi) [inline README.md]: update to Run on development env
also why u using </br>

File: index.js [hunk 5]
```diff
@@ -1,24 +1,127 @@
...
 })
 
-app.listen(PORT, () => `Listening at http://localhost:${PORT}`)
+app.post('/', async (req, res) => {
+  try {
+    if (req.body && req.body.type) {
+      switch(req.body.type) {
+        case 'ADDED_TO_SPACE':
+          return res.json({ 'text': `Thank you for adding me. Command *help* for more information` });
+        case 'MESSAGE':
```
Comment (gautamkrishnar) [inline index.js:28]: Can we add a check here to check if the messages actually matches the format of the commands supported by the bot.

Comment (gautamkrishnar) [reply index.js:28]: we can do that later as well, when we add more features

File: index.js [hunk 6]
```diff
@@ -1,24 +1,127 @@
...
+  }
+});
+
+// Start JBot functions
+/**
+ * Parses the text message according to what the text message is and responds accordingly
+ * @param {text message from user} argumentText 
+ * @returns Sanitized text which also sends the appropriate response
+ */
+async function textParser(argumentText) {
```
Comment (diwanshi) [inline index.js]: rename to parseText

File: index.js [hunk 7]
```diff
@@ -1,24 +1,127 @@
...
+  const issues = await Promise.all(issueTitles.map(issue => fetchJIRAInfo(issue)));
+  return issues;
+}
+
+/**
+ * Fetches JIRA information from the JIRA server
+ * @param {JIRA Issue number} issue 
+ * @returns Information about the JIRA - summary, status, reporter, assignee and priority
+ */
+async function fetchJIRAInfo(issue) {
```
Comment (diwanshi) [inline index.js]: rename to getJIRADetails

**Review by diwanshi — DISMISSED** [review]

File: index.js (hunk 5 above)
Comment (hybridx) [reply index.js:28]: It's added in the text parser

Comment (hybridx) [reply index.js:28]: We can add messge format check in the textParser

File: index.js [hunk 8]
```diff
@@ -1,24 +1,127 @@
...
+ * Parses the text message according to what the text message is and responds accordingly
+ * @param {text message from user} argumentText 
+ * @returns Sanitized text which also sends the appropriate response
+ */
+async function textParser(argumentText) {
+  const regex = /\b\w*[-']\w*\b/g;
+  const issueTitles = argumentText
+    .match(regex)
+    .map(word => word.toUpperCase());
+  const issues = await Promise.all(issueTitles.map(issue => fetchJIRAInfo(issue)));
```
Comment (gautamkrishnar) [inline index.js:54]: Add proper error handling here

File: README.md (hunk 2 above)
Comment (hybridx) [reply README.md]: Changed to 
```
A bot that retrieves the JIRA issue details from your JIRA server.
```

File: README.md (hunk 3 above)
Comment (hybridx) [reply README.md]: Updated

File: README.md (hunk 4 above)
Comment (hybridx) [reply README.md]: @diwanshi Sorry i didn't understood the later part 
```
also why you using ...
````

**Comment by gautamkrishnar** [comment]
@hybridx one more thing, i dmed the bot with `/help` command and it returned me this:

Screenshot 2021-05-12 at 7 54 02 PM

Not sure what happened, it should show unknown command or something.

File: index.js (hunk 6 above)
Comment (hybridx) [reply index.js]: Updated

File: index.js (hunk 7 above)
Comment (hybridx) [reply index.js]: Changed added

**Comment by hybridx** [comment]
> @hybridx one more thing, i dmed the bot with `/help` command and it returned me this:
> 
> Screenshot 2021-05-12 at 7 54 02 PM
> 
> Not sure what happened, it should show unknown command or something.

@gautamkrishnar  Yes I've closed the server for now and it's using the AppScript which has a static response

**Review by gautamkrishnar — APPROVED** [review]
👍🏻 LGTM

**Pull request merged**